File: frontend/src/services/payrollService.js
import axiosInstance from "../api/axiosConfig";

export const getPayrolls = () => {

    return axiosInstance.get("/payroll");

};

export const createPayroll = (payroll) => {

    return axiosInstance.post(
        "/payroll",
        payroll
    );

};

export const updatePayroll = (id, payroll) => {

    return axiosInstance.put(
        `/payroll/${id}`,
        payroll
    );

};

export const deletePayroll = (id) => {

    return axiosInstance.delete(
        `/payroll/${id}`
    );

};

export const getPayrollByEmployee = (employeeId) => {

    return axiosInstance.get(
        `/payroll/employee/${employeeId}`
    );

};

export const getPayrollByMonth = (month, year) => {

    return axiosInstance.get(
        `/payroll/month/${month}/${year}`
    );

};

File: backend/src/main/java/com/jejianil/erms/service/PayrollService.java
package com.jejianil.erms.service;

import com.jejianil.erms.dto.request.PayrollRequest;
import com.jejianil.erms.dto.response.PayrollResponse;

import java.util.List;

public interface PayrollService {

    PayrollResponse createPayroll(PayrollRequest request);

    List<PayrollResponse> getAllPayrolls();

    PayrollResponse getPayrollById(Long id);

    PayrollResponse updatePayroll(
            Long id,
            PayrollRequest request
    );

    void deletePayroll(Long id);

    List<PayrollResponse> getPayrollByEmployee(Long employeeId);

    List<PayrollResponse> getPayrollByMonth(
            Integer month,
            Integer year
    );
}